include/gridtools/common/for_each.hpp

```diff
@@ -54,7 +54,7 @@ namespace gridtools {
         } // namespace for_each_impl_
 
         template <class L>
-        constexpr for_each_impl_::for_each_f<L> for_each = {};
+        GT_TARGET constexpr for_each_impl_::for_each_f<L> for_each = {};
```

Comment:
I would prefer to limit the scope of the workaround and do the following:
```
template <class L>
#if defined(GT_TARGET_HAS_DEVICE) and  defined(__NVCC__)
GT_DEVICE
#endif
constexpr for_each_impl_::for_each_f<L> for_each = {};
```

Reply:
fine with me. done